assets/js/utils/render-frontend.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * External dependencies
+ */
+import { render } from 'react-dom';
+
+/**
+ * Renders a block component in the place of a specified set of selectors.
+ *
+ * @param {string}   selector   CSS selector to match the elements to replace.
+ * @param {function} Block      React block to use as a replacement.
+ * @param {function} [getProps] Function to generate the props object for the
+ * block.
+ */
+export default ( selector, Block, getProps = () => {} ) => {
+	const containers = document.querySelectorAll( selector );
+
+	if ( containers.length ) {
+		// Use Array.forEach for IE11 compatibility
+		Array.prototype.forEach.call( containers, ( el ) => {
```

Comment:
I'm curious, why not just `containers.forEach` which is functionally equivalent because `containers` is an array?  I see the comment about "for IE11" compatibility but won't babel transpile handle that?

Reply:
`containers` is a `NodeList` and IE11 doesn't support the `forEach` method:
https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach#Browser_Compatibility

I also thought Babel and the polyfills would take care of it, but it looks like they don't (just tested with IE11): [#](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/771#discussion_r308007214), [#](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/647#discussion_r297186203).

Reply:
> containers is a NodeList

Ahh right!  Thanks for verifying with polyfills too 👍